apps/deploy-web/src/components/new-deployment/ManifestEdit.tsx

```diff
@@ -200,12 +237,12 @@ export const ManifestEdit: React.FunctionComponent<Props> = ({ editedManifest, s
 
   const onModeChange = (mode: "yaml" | "builder") => {
     if (mode === selectedSdlEditMode) return;
-
     if (mode === "yaml") {
+      setEditedManifest(editedManifest as string);
```

Comment:
do we really need to cast here? this actually suggests that it can be null. This can cause issues. Something like this wouldn't need any type casts and would be actually type safe.

```typescript
  const onModeChange = (mode: "yaml" | "builder") => {
    if (mode === selectedSdlEditMode) return;
    const sdl = (mode === "yaml" && editedManifest) || sdlBuilderRef.current?.getSdl();

    if (sdl) {
      setEditedManifest(sdl);
    }

    setSelectedSdlEditMode(mode);
  };
```

Reply:
ok